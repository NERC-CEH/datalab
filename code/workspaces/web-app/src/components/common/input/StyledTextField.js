import React from 'react';
import TextField from '@material-ui/core/TextField';

const StyledTextField = props => (
  <TextField
    margin="dense"
    variant="outlined"
    {...props}
  />
);

export default StyledTextField;
