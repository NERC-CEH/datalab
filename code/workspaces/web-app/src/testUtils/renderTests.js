import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import theme from '../theme';

const AllTheProviders = ({ children }) => (
    <MuiThemeProvider theme={theme}>
        {children}
    </MuiThemeProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
