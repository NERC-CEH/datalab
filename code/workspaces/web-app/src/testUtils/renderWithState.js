import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import promise from 'redux-promise-middleware';
import {
  PROMISE_TYPE_PENDING,
  PROMISE_TYPE_SUCCESS,
  PROMISE_TYPE_FAILURE,
} from '../actions/actionTypes';

const promiseTypeSuffixes = [PROMISE_TYPE_PENDING, PROMISE_TYPE_SUCCESS, PROMISE_TYPE_FAILURE];

const TEST_USER = 'test-user';

export const buildDefaultTestState = () => ({
  authentication: {
    identity: {
      sub: TEST_USER,
    },
    permissions: {
      fetching: { inProgress: false },
      value: [],
    },
  },
  projectUsers: {
    fetching: { inProgress: false },
    value: [
      { userId: TEST_USER, role: 'admin' },
    ],
  },
});

export const renderWithState = (
  initialState,
  ComponentToRender,
  props = {},
  renderOptions = {},
) => {
  const mockStore = configureStore([promise({ promiseTypeSuffixes })]);
  const store = mockStore(initialState);

  const wrapper = render(
    <Provider store={store}>
      <ComponentToRender {...props} />
    </Provider>,
    renderOptions,
  );

  return {
    wrapper,
    store,
  };
};
