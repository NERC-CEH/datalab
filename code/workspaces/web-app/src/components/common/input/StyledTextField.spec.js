import React from 'react';
import { render } from '../../../testUtils/renderTests';
import StyledTextField from './StyledTextField';

jest.mock('@mui/material/TextField', () => props => (<div>TextField mock {JSON.stringify(props)}</div>));

const shallowRender = props => render(<StyledTextField {...props} />).container;
const expectShallowRenderToMatchSnapshot = props => expect(shallowRender(props)).toMatchSnapshot();

describe('StyledTextField', () => {
  it('renders to match snapshot when no extra props are provided', () => {
    expectShallowRenderToMatchSnapshot({});
  });

  it('renders to match snapshot forwarding passed props to child component', () => {
    expectShallowRenderToMatchSnapshot({ className: 'class-name', otherProp: 'value' });
  });

  it('renders to match snapshot overriding default prop value if provide different value', () => {
    expectShallowRenderToMatchSnapshot({ margin: 'normal' });
  });

  it('renders components passed as children', () => {
    const wrapper = render(
      <StyledTextField>
        <span>Child Content</span>
      </StyledTextField>,
    ).container;
    expect(wrapper).toMatchSnapshot();
  });
});
