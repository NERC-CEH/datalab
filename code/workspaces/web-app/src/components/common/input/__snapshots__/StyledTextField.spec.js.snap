// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyledTextField renders components passed as children 1`] = `
<WithStyles(ForwardRef(TextField))
  margin="dense"
  variant="outlined"
>
  <span>
    Child Content
  </span>
</WithStyles(ForwardRef(TextField))>
`;

exports[`StyledTextField renders to match snapshot forwarding passed props to child component 1`] = `
<WithStyles(ForwardRef(TextField))
  className="class-name"
  margin="dense"
  otherProp="value"
  variant="outlined"
/>
`;

exports[`StyledTextField renders to match snapshot overriding default prop value if provide different value 1`] = `
<WithStyles(ForwardRef(TextField))
  margin="normal"
  variant="outlined"
/>
`;

exports[`StyledTextField renders to match snapshot when no extra props are provided 1`] = `
<WithStyles(ForwardRef(TextField))
  margin="dense"
  variant="outlined"
/>
`;
